app/src/main/java/com/isc/hermes/controller/MapWayPointController.java

```diff
@@ -49,23 +63,42 @@ public boolean onMapClick(@NonNull LatLng point) {
         } catch (JSONException e) {
             throw new RuntimeException(e);
         }
+        TrafficUploader.getInstance().setLastClickedPoint(point);
+
         return true;
     }
 
     /**
      * Method to perform action to click on map
      * @param point Is point passed as parameter with its latitude and longitude
      */
-    private void doMarkOnMapAction(LatLng point){
-        if (isMarked){
+
+    private void doMarkOnMapAction(LatLng point) {
```

Comment:
You should refactor this method, there are parts of code that are repeated and can go to the other method

Reply:
Done